src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollListBoxAutomationPeer.cs

```diff
@@ -26,8 +28,19 @@ protected override List<AutomationPeer> GetChildrenCore()
             {
                 await infiniteScrollListBox.ItemsLock.ExecuteAsync(() =>
                 {
-                    // Don't return the LoadingStatusIndicator as an AutomationPeer, otherwise narrator will report it as an item in the list of packages, even when not visible
-                    value = base.GetChildrenCore()?.Where(lbiap => !(((ListBoxItemAutomationPeer)lbiap).Item is LoadingStatusIndicator)).ToList() ?? new List<AutomationPeer>();
+                    List<AutomationPeer> items = base.GetChildrenCore();
+                    if (items == null)
+                    {
+                        value = new List<AutomationPeer>();
+                    }
+                    else
+                    {
+                        var listCollectionView = (ListCollectionView)CollectionViewSource.GetDefaultView(items);
+                        // Don't return the LoadingStatusIndicator as an AutomationPeer, otherwise narrator will report it as an item in the list of packages, even when not visible
+                        IEnumerable<AutomationPeer> packageItemsInView = items.Where(item => ((ListBoxItemAutomationPeer)item).Item is PackageItemListViewModel
```

Comment:
```suggestion
                        IEnumerable<AutomationPeer> packageItemsInView = items.Where(item => ((ListBoxItemAutomationPeer)item).Item is PackageItemListViewModel &&
                                                                                        listCollectionView.Contains(item));

```

Reply:
Honestly GH makes it so difficult haha. I tried to delete 16 spaces (I did in my editor) and copy paste here. Try it out, but I'm not gonna sweat over it.

Reply:
That didn't work 😄 ... good to know!